src/document.c

```diff
@@ -1017,15 +1017,18 @@ static gboolean load_text_file(const gchar *locale_filename, const gchar *displa
 
 	if (filedata->readonly)
 	{
-		const gchar *warn_msg = _(
-			"The file \"%s\" could not be opened properly and has been truncated. " \
-			"This can occur if the file contains a NULL byte. " \
-			"Be aware that saving it can cause data loss.\nThe file was set to read-only.");
+		gchar *warn_msg = g_strdup_printf(_(
+			"The file \"%s\" could not be opened properly and has been truncated. "
+			"This can occur if the file contains a NULL byte. "
+			"Be aware that saving it can cause data loss.\nThe file was set to read-only."),
+			display_filename);
 
 		if (main_status.main_window_realized)
-			dialogs_show_msgbox(GTK_MESSAGE_WARNING, warn_msg, display_filename);
+			dialogs_show_msgbox(GTK_MESSAGE_WARNING, "%s", warn_msg);
+
+		ui_set_statusbar(TRUE, "%s", warn_msg);
```

Comment:
Maybe the format should be taken out of `ui_set_statusbar` and `msgwin_status_add` or the compiler told they are [formats](https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Function-Attributes.html) [and](https://releases.llvm.org/9.0.0/tools/clang/docs/AttributeReference.html#format).

Note this is passed through printf substitution here, in `ui_set_statusbar()` and `msgwin_status_add()` called from that.

Reply:
> Maybe the format should be taken out of `ui_set_statusbar` and `msgwin_status_add`

`ui_set_statusbar()` has many callers using the formatting feature, and I don't think it's worth adding a specific variant not taking a format but a plain string (quick grep gives me 12 out of 87 calls pass `"%s"` using this branch).
OTOH we have [`msgwin_status_add_string()`](https://github.com/geany/geany/blob/eef67ecc200a6cf07f277c28282facf9854cd2ea/src/msgwindow.h#L62), which I use in the patch.

> or the compiler told they are [formats](https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Function-Attributes.html) [and](https://releases.llvm.org/9.0.0/tools/clang/docs/AttributeReference.html#format).

What [do](https://docs.gtk.org/glib/func.GNUC_PRINTF.html) [you](https://github.com/geany/geany/blob/eef67ecc200a6cf07f277c28282facf9854cd2ea/src/ui_utils.h#L108) [mean](https://github.com/geany/geany/blob/eef67ecc200a6cf07f277c28282facf9854cd2ea/src/msgwindow.h#L60)?

Reply:
> What [do](https://docs.gtk.org/glib/func.GNUC_PRINTF.html) [you](https://github.com/geany/geany/blob/eef67ecc200a6cf07f277c28282facf9854cd2ea/src/ui_utils.h#L108) [mean](https://github.com/geany/geany/blob/eef67ecc200a6cf07f277c28282facf9854cd2ea/src/msgwindow.h#L60)?

Oh, its only on the declaration, I looked at the [definition](https://github.com/geany/geany/blob/eef67ecc200a6cf07f277c28282facf9854cd2ea/src/ui_utils.c#L167), ok.

> which I use in the patch.

which patch?

My point was that `ui_set_statusbar` printfs the input then calls `msgwin_status_add` with `%s` to re-printf the result string that it just printfed ((mis)using printf as a verb), so maybe that should use `msgwin_status_add_string()`?

> 12 out of 87 calls pass "%s"

Not inconsiderable.  In this case you need to printf the string before sending it to the status bar because it is used in the dialog, but since status bar is always doing GTK with the result the cost of re-printfing possibly is immaterial.

Reply:
> which patch?

https://github.com/geany/geany/pull/3665/commits/14d4bb0e05b85024db18349e4b86112940c7eeeb#diff-d5920a54fc6fff9b49052b09ca50c0608cf55cc93b5a46713bf8a4de1132eed0R1891 but that's irrelevant given I didn't understand what you meant, see below.

> My point was that `ui_set_statusbar` printfs the input then calls `msgwin_status_add` with `%s` to re-printf the result string that it just printfed ((mis)using printf as a verb), so maybe that should use `msgwin_status_add_string()`?

Ooh, OK.  Indeed, `ui_set_statusbar()` should call `msgwin_status_add_string()` instead of `msgwin_status_add()`.  I can add (pun intended) it here, although it's kind of an unrelated optimization.

> > 12 out of 87 calls pass "%s"
> 
> Not inconsiderable. In this case you need to printf the string before sending it to the status bar because it is used in the dialog, but since status bar is always doing GTK with the result the cost of re-printfing possibly is immaterial.

Well, I can do an "optimize printf usage" pass and introduce `ui_set_statusbar_string()`, but as you say I believe it's not gonna matter at the performance level (handling format `"%s"` is likely pretty trivial, as would be handling formats without placeholders).